openpype/hosts/maya/api/plugin.py

```diff
@@ -143,14 +143,45 @@ def load(
         assert os.path.exists(self.fname), "%s does not exist." % self.fname
 
         asset = context['asset']
+        subset = context['subset']
+        settings = get_project_settings(os.environ['AVALON_PROJECT'])
         loaded_containers = []
 
         count = options.get("count") or 1
+
         for c in range(0, count):
-            namespace = namespace or lib.unique_namespace(
-                "{}_{}_".format(asset["name"], context["subset"]["name"]),
-                prefix="_" if asset["name"][0].isdigit() else "",
-                suffix="_",
+            custom_naming = settings['maya']['load']['reference_loader']['naming']  # noqa
+            group_name = None
+
+            if ':' not in custom_naming:
+                raise ValueError(
+                    "Wrong format for namespace, missing ':' separator"
+                )
+            elif custom_naming.strip()[0] == ':':
+                raise ValueError(
+                    "Wrong format for namespace, \
+missing content before ':' separator"
+                )
+            elif custom_naming.strip()[-1] == ':':
+                raise ValueError(
+                    "Wrong format for namespace, \
+missing content after ':' separator"
+                )
+
+            custom_naming = custom_naming.format(
```

Comment:
I don't think it's good idea to give option to use any data in asset document and subset document. There should be defined subset of keys which should ideally match the rest of current templates.
```
{
    "asset": asset["name"],
    "subset": subset["name"],
    "family": subset["data"].get("family") or subset["data"]["families"][0]
}
```

e.g. `asset["type"]` will always be `"asset"`. Who set the settings should not know that family is in `subset["data"]["family"]`.